### ISSUE TITLE
no-parent-name-construct-id-match: misses assignments, else branches, loops, and non-first declarators

### ISSUE THREAD

**Issue by ren-yamanashi** [ren-yamanashi/eslint-cdk-plugin#535]
### Describe the bug

`no-parent-name-construct-id-match` walks constructor statements with a hand-written traversal that only understands a few statement shapes. `new` expressions in the following (very common) positions are never validated:

1. assignment form: `this.bucket = new Bucket(this, "MyConstruct")` (an `ExpressionStatement` whose expression is an `AssignmentExpression`, not a `NewExpression`)
2. the `else` branch of an `if` statement (only `statement.consequent` is traversed)
3. loop bodies (`for` / `for...of` / `while` etc. — no switch case matches, statement is skipped)
4. second and later declarators (`const a = 1, b = new Bucket(...)` — only `declarations[0].init` is checked)

**Repro:**

```ts
import { Bucket } from "aws-cdk-lib/aws-s3";
import { Construct } from "constructs";

export class AssignParent extends Construct {
  public readonly bucket: Bucket;
  constructor(scope: Construct, id: string) {
    super(scope, id);
    this.bucket = new Bucket(this, "AssignParent"); // 1. not reported
  }
}

export class BranchParent extends Construct {
  constructor(scope: Construct, id: string, props: { multiAz: boolean }) {
    super(scope, id);
    if (props.multiAz) {
      new Bucket(this, "BranchParent"); // control: reported
    } else {
      new Bucket(this, "BranchParent"); // 2. not reported
    }
  }
}

export class LoopParent extends Construct {
  constructor(scope: Construct, id: string) {
    super(scope, id);
    for (let i = 0; i < 1; i++) {
      new Bucket(this, "LoopParent"); // 3. not reported
    }
  }
}

export class MultiDeclParent extends Construct {
  constructor(scope: Construct, id: string) {
    super(scope, id);
    const marker = "x",
      bucket = new Bucket(this, "MultiDeclParent"); // 4. not reported
    void marker;
    void bucket;
  }
}
```

### Expected Behavior

All four IDs match their parent class name and should be reported, exactly like the direct-statement form.

### Current Behavior

Only the control (consequent branch) is reported:

```
repro/no-parent-name-construct-id-match/branches.ts         9:24  error  Construct ID 'BranchParent' should not match parent construct name 'BranchParent'. Use a more specific identifier  awscdk/no-parent-name-construct-id-match
repro/no-parent-name-construct-id-match/this-assignment.ts  8:22  error  Construct ID 'ControlParent' should not match parent construct name 'ControlParent'. Use a more specific identifier  awscdk/no-parent-name-construct-id-match
```

(`this-assignment.ts:8` is a direct-statement control in the same file; the `this.bucket = ...` case at line 17 is not reported.) Oxlint reports the identical set.

### Plugin

eslint-plugin-awscdk, oxlint-plugin-awscdk

### Plugin Version

eslint-plugin-awscdk 4.4.0, oxlint-plugin-awscdk 1.0.0

### Language Version

TypeScript 6.0.3 (eslint-plugin-awscdk) / TypeScript 7 preview via oxlint-tsgolint (oxlint-plugin-awscdk)

### Additional Context

- Verified on `main` (35b889a), macOS, rule-isolated configs.
- Root cause: `packages/*/src/rules/no-parent-name-construct-id-match.ts` — `validateConstructorBody` / `traverseStatements` / `validateStatement` switch on a closed set of statement types (`VariableDeclaration` first declarator, `ExpressionStatement` with a direct `NewExpression`, `IfStatement.consequent`, `SwitchStatement`).
- Every other rule in the suite uses a `NewExpression` visitor and does not have this blind spot; converting this rule to the same visitor-based approach would remove the whole class of gaps (including `new` inside call arguments, `return new ...`, etc.).
- `this.bucket = new Bucket(this, id)` is the single most common way to expose a child construct, so gap 1 in particular defeats the rule in typical code.
